components/director/internal/repo/update.go

```diff
@@ -126,21 +176,28 @@ func (u *universalUpdater) unsafeUpdateSingleWithFields(ctx context.Context, dbE
 		return err
 	}
 
-	query, err := u.buildQuery(fieldsToSet, isGlobal)
+	query, err := u.buildQuery(fieldsToSet, tenant, resourceType)
 	if err != nil {
 		return err
 	}
 
+	if entityWithExternalTenant, ok := dbEntity.(EntityWithExternalTenant); ok && (u.tenantColumn == nil && len(tenant) > 0) {
+		dbEntity = entityWithExternalTenant.DecorateWithTenantID(tenant)
+	}
+
 	log.C(ctx).Debugf("Executing DB query: %s", query)
 	res, err := persist.NamedExecContext(ctx, query, dbEntity)
-	if err = persistence.MapSQLError(ctx, err, u.resourceType, resource.Update, "while updating single entity from '%s' table", u.tableName); err != nil {
+	if err = persistence.MapSQLError(ctx, err, resourceType, resource.Update, "while updating single entity from '%s' table", u.tableName); err != nil {
 		return err
 	}
 
 	affected, err := res.RowsAffected()
 	if err != nil {
 		return errors.Wrap(err, "while checking affected rows")
 	}
+	if affected == 0 && (len(tenant) > 0 || u.tenantColumn != nil) {
```

Comment:
This way you will just hide the boolean expression in a method which should either way be opened to see what it is doing. I don't think that the name and the arguments provided will give more clarity as the expression is easy enough anyways.

I think that instead of func we at-least can create a Boolean variable above the if, but still don't see any benefit, do you think it is not clear enough and naming the expression somehow will help?

Reply:
Mixing && and || in one expression can be confusing especially in case you are not confident on the concept - when use externally provided tenant and when the embeded one and after that when you should say the issue is that you dont have the rights to perform the operation or the issue is somwhere else. It is not just the straightforward read and understand but you should put some thought into it. Maybe boolean variable named checkedAuthorization or usedAuthorization will make it easier to understand.